src/saveload/saveload.h

```diff
@@ -401,15 +400,45 @@ enum ChunkType {
 /** Handlers and description of chunk. */
 struct ChunkHandler {
 	uint32 id;                          ///< Unique ID (4 letters).
-	ChunkSaveLoadProc *save_proc;       ///< Save procedure of the chunk.
-	ChunkSaveLoadProc *load_proc;       ///< Load procedure of the chunk.
-	ChunkSaveLoadProc *ptrs_proc;       ///< Manipulate pointers in the chunk.
-	ChunkSaveLoadProc *load_check_proc; ///< Load procedure for game preview.
 	ChunkType type;                     ///< Type of the chunk. @see ChunkType
+
+	ChunkHandler(uint32 id, ChunkType type) : id(id), type(type) {}
+
+	virtual ~ChunkHandler() {}
+
+	/**
+	 * Save the chunk.
+	 * Must be overriden, unless Chunk type is CH_READONLY.
+	 */
+	virtual void Save() const { NOT_REACHED(); }
+
+	/**
+	 * Load the chunk.
+	 * Must be overriden.
+	 */
+	virtual void Load() const = 0;
+
+	/**
+	 * Fix the pointers.
+	 * Pointers are saved using the index of the pointed object.
+	 * On load, pointers are filled with indices and need to be fixed to point the the real object.
```

Comment:
`the the` -> `to the` on this line.
`overriden` -> `overridden` in all these functions.